### PR TITLE
[boost] update to 1.79.0

### LINKED ISSUE #24127: [boost] update to 1.79
Library name: boost 

New version number: 1.79

Other information that may be useful (release notes, etc...)
https://github.com/boostorg/boost/tree/boost-1.79.0
https://www.boost.org/users/history/version_1_79_0.html

( Asio: added OpenSSL 3.0 compatibility... )

### PR DESCRIPTION
**Pull request by yurybura**
Updates Boost to version 1.79.0:
https://www.boost.org/users/history/version_1_79_0.html

Fixes #24127

### CODE REVIEW

File: ports/boost-wave/portfile.cmake [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
+    SHA512 d573202b2eee3d2810c0cbbbffdb61dea957f8a4be33755a2e797fd23ae531a61a66a3b2cf1fa1d114781b72d280db8a45e350c437dc57d9bed1954c596a83c2
     HEAD_REF master
 )
 
-if(NOT DEFINED CURRENT_HOST_INSTALLED_DIR)
-    message(FATAL_ERROR "boost-wave requires a newer version of vcpkg in order to build.")
-endif()
+file(READ "${SOURCE_PATH}/build/Jamfile.v2" _contents)
+string(REPLACE "import ../../config/checks/config" "import ../config/checks/config" _contents "${_contents}")
+file(WRITE "${SOURCE_PATH}/build/Jamfile.v2" "${_contents}")
```
Comment (JonLiu1993) [inline ports/boost-wave/portfile.cmake:13]:
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/build/Jamfile.v2" "import ../../config/checks/config" "import ../config/checks/config")
```

File: scripts/boost/post-source-stubs/wave.cmake [hunk 2]
```diff
@@ -0,0 +1,4 @@
+file(READ "${SOURCE_PATH}/build/Jamfile.v2" _contents)
+string(REPLACE "import ../../config/checks/config" "import ../config/checks/config" _contents "${_contents}")
+file(WRITE "${SOURCE_PATH}/build/Jamfile.v2" "${_contents}")
```
Comment (JonLiu1993) [inline scripts/boost/post-source-stubs/wave.cmake:3]:
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/build/Jamfile.v2" "import ../../config/checks/config" "import ../config/checks/config")
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/boost-wave/portfile.cmake (hunk 1 above)
Comment (yurybura) [reply ports/boost-wave/portfile.cmake]: Done.

File: scripts/boost/post-source-stubs/wave.cmake (hunk 2 above)
Comment (yurybura) [reply scripts/boost/post-source-stubs/wave.cmake]: Done.

**Comment by JonLiu1993** [comment]
LGTM! thanks for your contribution!

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thank you!

**Pull request merged**